packages/gasket-react-intl/src/index.d.ts

```diff
@@ -34,10 +34,10 @@ export function withLocaleRequired<Props>(localePathPart?: LocalePathPart, optio
  * @param [props.loading] - Custom component to show while loading
  * @returns element
  */
-export function LocaleRequired(props: {
+export function LocaleRequired(props: PropsWithChildren<{
   localesPath: LocalePathPart;
   loading?: React.ReactNode;
-}): React.ReactNode | null;
+}>): React.ReactElement | null;
```

Comment:
Maybe something like this is best, to explicitly make it a component type:

```
export const LocaleRequired: React.ComponentType<
  PropsWithChildren<{
    localesPath: LocalePathPart;
    loading?: React.ReactNode;
  }>
>;

Reply:
Yes, using ComponentType could work too.

Change was because of:
```
'LocaleRequired' cannot be used as a JSX component.
  Its return type 'ReactNode' is not a valid JSX element.
    Type 'undefined' is not assignable to type 'Element | null'.ts(2786)
```